yroots/subdivision.py

```diff
@@ -898,7 +900,7 @@ def subdivision_solve_1d(f,a,b,deg,target_deg,interval_data,root_tracker,tols,ma
     # Trim the coefficient array (reduce the degree) as much as we can.
     # This identifies a 'good degree' with which to approximate the function
     # if it is less than the given approx degree.
-    while np.isclose(0, coeff[-1]):
+    while np.isclose(0, coeff[-1], atol=tols.abs_approx_tol, rtol=tols.rel_approx_tol):
```

Comment:
Hmm thanks for noticing this error, but this probably should be even more rigorous. Unfortunately, I think that might mean changing the structure of the code. Here's some tentative code:

`# Approximate the function using Chebyshev polynomials
    coeff, inf_norm = interval_approximate_1d(f,a,b,deg)

    good_deg = max(len(coeff) - 1, 1)

    # coeff, inf_norm = interval_approximate_1d(f,a,b,good_deg)
    coeff2, inf_norm = interval_approximate_1d(f,a,b,good_deg*2,inf_norm)

    coeff2[slice_top(coeff)] -= coeff

    # Calculate the approximate error between the deg and 2*deg approximations
    error = np.sum(np.abs(coeff2))

    allowed_error = tols.abs_approx_tol+tols.rel_approx_tol*inf_norm

    if error > allowed_error:
        # Subdivide the interval and recursively call the function.
        div_spot = a + (b-a)*RAND
        subdivision_solve_1d(f, a, div_spot, good_deg, target_deg,interval_data,root_tracker,tols,max_level,level+1)
        subdivision_solve_1d(f, div_spot, b, good_deg, target_deg,interval_data,root_tracker,tols,max_level,level+1)
    else:
        # Trim the coefficient array (reduce the degree) as much as we can.
        # This identifies a 'good degree' with which to approximate the function
        # if it is less than the given approx degree.
        last_coeff_size = abs(coeff[-1])
        new_error = error + last_coeff_size
        while new_error < allowed_error:
            if len(coeff) == 1:
                break
            error = new_error
            #maybe a list pop here? idk if worth it to switch away from arrays
            coeff = coeff[:-1]
            last_coeff_size = abs(coeff[-1])
            new_error = error + last_coeff_size

        # Run interval checks to eliminate regions`